src/App.tsx

```diff
@@ -1,42 +1,81 @@
-import { useState, useMemo } from 'react';
-import { Github, Loader } from 'lucide-react';
-import { RepoCard } from './components/RepoCard';
-import { SearchBar } from './components/SearchBar';
-import { SortSelect } from './components/SortSelect';
-import { useGitHub } from './hooks/useGitHub';
-import { SortOption } from './types';
-import { useDispatch, useSelector } from 'react-redux';
-import { setRepos as setReposInRedux } from './redux/repoSlice';
-import { RootState } from './redux/store';
+import { Github, Loader } from "lucide-react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { FilteringOptions } from "./components/FilteringOptions";
+import { RepoCard } from "./components/RepoCard";
+import { SearchBar } from "./components/SearchBar";
+import { SortSelect } from "./components/SortSelect";
+import { useGitHub } from "./hooks/useGitHub";
+import { setRepos as setReposInRedux } from "./redux/repoSlice";
+import { RootState } from "./redux/store";
+import { Repository, SortOption } from "./types";
+import { FilterState } from "./redux/filteringSlice";
 
 const sortOptions: SortOption[] = [
-  { label: 'Least Recently Updated', value: 'updated', direction: 'asc' },
-  { label: 'Recently Updated', value: 'updated', direction: 'desc' },
-  { label: 'Least Pull Requests', value: 'pulls', direction: 'asc' },
-  { label: 'Most Pull Requests', value: 'pulls', direction: 'desc' },
-  { label: 'Least Stars', value: 'stars', direction: 'asc' },
-  { label: 'Most Stars', value: 'stars', direction: 'desc' },
+  { label: "Least Recently Updated", value: "updated", direction: "asc" },
+  { label: "Recently Updated", value: "updated", direction: "desc" },
+  { label: "Least Pull Requests", value: "pulls", direction: "asc" },
+  { label: "Most Pull Requests", value: "pulls", direction: "desc" },
+  { label: "Least Stars", value: "stars", direction: "asc" },
+  { label: "Most Stars", value: "stars", direction: "desc" },
 ];
 
 function App() {
-  const { repos, loading, error, sortRepos, searchRepos } = useGitHub();
-  const [searchTerm, setSearchTerm] = useState('');
+  const { repos, loading, error, fetchRepos } = useGitHub();
+  const [searchTerm, setSearchTerm] = useState("");
   const [selectedSort, setSelectedSort] = useState<SortOption>(sortOptions[0]);
+  const filterState = useSelector((state: RootState) => state.filtering);
   const reposRedux = useSelector((state: RootState) => state.repo.value);
-  
-  // Take the Repos Returned from the Github Hook, and set them in Redux
   const dispatch = useDispatch();
-  dispatch(setReposInRedux(repos));
+
```

Comment:
Make a comment here about why this setup is needed instead of using the `repos` from useGithub.